1/incrementing-the-value.md

```diff
@@ -12,39 +12,17 @@ If you are simply _reading_ from the contract storage, you only need to pass `&s
 ```rust
 impl MyContract {
     #[ink(message)]
-    fn my_getter(&self) -> u32 {
-        *self.my_number
+    pub fn my_getter(&self) -> u32 {
+        self.my_number
     } 
 
     #[ink(message)]
-    fn my_setter(&mut self, new_value: u32) {
-        self.my_number.set(new_value);
+    pub fn my_setter(&mut self, new_value: u32) {
+        self.my_number = new_value;
     }
 }
 ```
 
-## Modifying a Storage Value
-
-You can always update the value of a storage item by calling `set` again.
-
-However, if you know the value is already set, then you can modify the value in a more ergonomic way:
-
-```rust
-impl MyContract {
-    #[ink(message)]
-    fn my_setter(&mut self, new_value: u32) {
-        self.my_number.set(new_value);
-    }
-
-    #[ink(message)]
-    fn my_adder(&mut self, add_value: u32) {
-        self.my_number += add_value;
-    }
-}
-```
-
-However, if the value is not initialized, your contract will compile fine, but will panic during contract execution! **We really cannot understate how easy it is to make mistakes this way.**
-
```

Comment:
This should still be documented but using the new type `ink_storage::lazy::Lazy<T>` which shares some similarities with the old `ink_core::storage::Value<T>`.

Reply:
You use `Lazy` to lazily manipulate the underlying value. So if you use a raw `i32` in your storage the contract will load this value upon contract execution directly even if you never use it. A `Lazy<i32>` only loads it if it is actually used. For flipper and incrementer contracts we use `i32` instead of `Lazy<i32>` since all their methods always operate on the `value` so using a `Lazy<i32>` instead wouldn't provide us with any benefit. Also `Lazy<T>` has some minor overhead compared to using `T` directly.